pkg/kubectl/printers/flags.go

```diff
@@ -0,0 +1,110 @@
+package printers
+
+import (
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+// PrinterProvider is capable of providing a printer
+// based on a given set of options
+type PrinterProvider interface {
+	AddFlags(cmd *cobra.Command)
+	ToPrinter(outputFormat string) (printer printers.ResourcePrinter, matchedPrinter bool, err error)
+}
+
+// PrintFlags composes all known printer flag structs
+// and provides a method of retrieving a known printer
+// based on flag values provided.
+type PrintFlags struct {
+	// embedded structs (so that calling commands can bind values)
+	CustomColumnsPrintFlags *CustomColumnsPrintFlags
+	HumanPrintFlags         *HumanPrintFlags
```

Comment:
No.  This isn't a general printer.  This is a printer just for `get.go`.  How about making it a lot more obvious by first moving `get.go` to `pkg/kubectl/cmd/get/get.go` and then start segregating `get.go` only pieces.